### PR TITLE
[SMALLFIX] Add debug log on BlockMaster request size

### PR DESCRIPTION
**Pull request by jiacheliu3**
When the worker has a large number of blocks the request can be big for worker register and block heartbeat. Adding a little debug log for better traffic visibility.

### CODE REVIEW

**Comment by jiacheliu3** [comment]
@bf8086 anther alternative is to check the message size, and log a warn when this is larger than a threshold of, say 10MB.

**Comment by gpang** [comment]
Why do we need to know the sizes of these messages?

**Comment by jiacheliu3** [comment]
> Why do we need to know the sizes of these messages?

@gpang Some user environments have a huge number of blocks per worker. For example, a worker with 6TB storage in total can have more than 2 million blocks. On registering with the master we see much stress in CPU and network. So I'd like to get more visibility to better understand.

File: core/server/master/src/main/java/alluxio/master/block/BlockMasterWorkerServiceHandler.java [hunk 1]
```diff
@@ -61,6 +61,12 @@ public BlockMasterWorkerServiceHandler(BlockMaster blockMaster) {
   @Override
   public void blockHeartbeat(BlockHeartbeatPRequest request,
       StreamObserver<BlockHeartbeatPResponse> responseObserver) {
+    if (LOG.isDebugEnabled()) {
```
Comment (maobaolong) [inline core/server/master/src/main/java/alluxio/master/block/BlockMasterWorkerServiceHandler.java:64]: Is this log level check necessary?

**Comment by bf8086** [comment]
> @bf8086 anther alternative is to check the message size, and log a warn when this is larger than a threshold of, say 10MB.

yeah you can probably get a rough estimation by checking the block counts instead which has less performance impact.

**Review by bf8086 — APPROVED** [review]

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

File: core/server/master/src/main/java/alluxio/master/block/BlockMasterWorkerServiceHandler.java (hunk 1 above)
Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/block/BlockMasterWorkerServiceHandler.java:64]: I think we avoid one calculation if the log level is above debug

**Pull request merged**